### PR TITLE
{bio}[foss/2018b] SAMtools 1.9, BCFtools 1.9, HTSlib 1.9, BEDTools 2.27.1, BamTools 2.5.1

### PR DESCRIPTION
**Pull request by fizwit**
BioInformatics tools for 2018b

BCFtools-1.9-foss-2018b.eb
BEDTools-2.27.1-foss-2018b.eb
BamTools-2.5.1-foss-2018b.eb
HTSlib-1.9-foss-2018b.eb
SAMtools-1.9.0-foss-2018b.eb

### CODE REVIEW

File: easybuild/easyconfigs/b/BCFtools/BCFtools-1.9-foss-2018b.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/samtools/%(namelower)s/releases/download/%(version)s']
+sources = [SOURCELOWER_TAR_BZ2]
+checksums = ['6f36d0e6f16ec4acf88649fb1565d443acf0ba40f25a9afd87f14d14d13070c8']
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('HTSlib', '1.9'),
+    ('GSL', '2.5', '-GCCcore-7.3.0', True),
```
Comment (boegel) [inline easybuild/easyconfigs/b/BCFtools/BCFtools-1.9-foss-2018b.eb]: @fizwit No need to hardcode to `GCCcore` here (that's also what's breaking the tests):

```python
('GSL', '2.5'),
```

File: easybuild/easyconfigs/b/BEDTools/BEDTools-2.27.1-foss-2018b.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = ['https://github.com/arq5x/bedtools2/releases/download/v%(version)s/']
+sources = ['bedtools-%(version)s.tar.gz']
+checksums = ['c8c2acbaf3f9cb92dcf8e5cd59af6b31ae9c4598efb786ba6c84f66ca72fafd9']
+
+buildopts = 'CXX="$CXX"'
+
+dependencies = [
+    ('BamTools', '2.5.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/b/BEDTools/BEDTools-2.27.1-foss-2018b.eb]: @fizwit The existing `BEDTools` easyconfig also has a dependency on `zlib`, please include it here too next to `BamTools`

Comment (fizwit) [reply easybuild/easyconfigs/b/BEDTools/BEDTools-2.27.1-foss-2018b.eb]: ok

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2094.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4131fded4ace528e68dc25f5d17db429 for a full test report.

File: easybuild/easyconfigs/b/BCFtools/BCFtools-1.9-foss-2018b.eb (hunk 1 above)
Comment (fizwit) [reply easybuild/easyconfigs/b/BCFtools/BCFtools-1.9-foss-2018b.eb]: Easybuild should resolve lower toolchains, but sometimes it does not work.
EasyBuild/3.6.2
ERROR: Build of /home/neo/BCFtools-1.9-foss-2018b.eb failed (err: 'build failed (first 300 chars): Missing modulesfor one or more dependencies: GSL/2.5-foss-2018b')
outside_user@1b732d5d33d1:/home/neo$ ml avail GSL

------------------------------------------------ /app/modules/all -------------------------------------------------
   GSL/2.3-foss-2016b    GSL/2.5-GCCcore-7.3.0 (D)

Comment (boegel) [reply easybuild/easyconfigs/b/BCFtools/BCFtools-1.9-foss-2018b.eb]: @fizwit This is related to a known problem, see https://github.com/easybuilders/easybuild-framework/issues/2375 .

The dependency resolution mechanism currently requires that the `GSL-2.5-GCCcore-7.3.0.eb` easyconfig file is included in the robot search path; if it's not there, only `GSL/2.5-foss-2018b` (i.e. the top-level toolchain) is considered.

I hope to find some time to try and fix that bug soon...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2619.swalot.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/c256bb7b2b41f4050f6fdb0e5e059a2c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (5 easyconfigs in this PR)
node3151.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/49b8652426712bbb88b0563329b852e2 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @fizwit!

**Pull request merged**